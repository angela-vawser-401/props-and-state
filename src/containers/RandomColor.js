import React, { Component } from 'react';
import ColorDisplay from '../components/colors/ColorDisplay';

// Changes color randomly every second

export default class RandomColor extends Component {
  state = {
    backgroundColor: 'black' //RandomColor
  };

  selectColor = () => {
    const colors = ['#75d5fd', '#b76cfd', 'black', '#ffacfc', '#f148fb', '#7122fa', '#560a86', '#08f7fe', '#09fbd3', '#fe53bb', '#f5d300', '#011ffd', '#ff2281', '#ff9472', '#3b55ce'];
    const random = Math.floor(Math.random() * colors.length);
    this.setState({ backgroundColor: colors[random] });
  }

  componentDidMount() {
    this.selectColor();
    setInterval(this.selectColor, 1000);
  }

  render() {
    const { backgroundColor } = this.state;

    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
      </>
    );
  }
}
