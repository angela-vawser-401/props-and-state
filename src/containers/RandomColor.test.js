import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('Random Color Component', () => {
  it('renders Random Color', () => {
    const wrapper = shallow(<RandomColor />);
    wrapper.setState({ backgroundColor: 'black' });
    expect(wrapper).toMatchSnapshot();
  });
});
