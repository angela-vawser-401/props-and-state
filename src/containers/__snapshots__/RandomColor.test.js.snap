// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Color Component renders Random Color 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="black"
  />
</Fragment>
`;
